packages/esm-appointments-app/src/form/appointments-form.component.tsx

```diff
@@ -725,7 +729,9 @@ function TimeAndDuration({ isTablet, t, watch, control, services }) {
           render={({ field: { onChange, value } }) => (
             <TimePicker
               id="time-picker"
-              pattern="([\d]+:[\d]{2})"
+              pattern={time12HourFormatRegexPattern}
+              invalid={!isValidTime(value)}
+              invalidText={t('invalidTime', 'Invalid time')}
```

Comment:
I don't quite know what the `pattern` property here accomplishes.  It doesn't seem to have much effect.  The `invalid` and `invalidText` seem to be what is needed to display a validation error below the field if the input value is invalid.

Reply:
fwiw, the pattern attribute is explained here: https://react.carbondesignsystem.com/?path=/docs/components-timepicker--overview

On quick glance, it seems like it is *supposed* to do what the invalid function is doing?  I have found that React Carbon elements don't have the greatest documentation (in this case, perhaps the pattern works in conjunction with warning, just a guess?) so if we can't get this to work and are relying on invalid, probably just remove the pattern attribute entirely to avoid future confusion.

Reply:
`pattern` seems to be what the browser uses to check pattern on form submission. https://www.w3schools.com/TAGS/att_input_pattern.asp

I'm guess `invalid` can be set whenever, and doesn't need to be tied to form submission action.